packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/FLTCam.m

```diff
@@ -1199,6 +1199,64 @@ - (void)setZoomLevel:(CGFloat)zoom withCompletion:(void (^)(FlutterError *_Nulla
   completion(nil);
 }
 
+- (void)setVideoStabilizationMode:(FCPPlatformVideoStabilizationMode)mode
+                   withCompletion:(void (^)(FlutterError *_Nullable))completion {
+  AVCaptureVideoStabilizationMode avMode = getAvCaptureVideoStabilizationMode(mode);
+  bool isSupported = [self isVideoStabilizationAvModeSupported:avMode];
+
+  if (!isSupported) {
+    completion([FlutterError errorWithCode:@"VIDEO_STABILIIZATION_ERROR"
+                                   message:@"Unavailable video stabilization mode."
+                                   details:nil]);
+    return;
+  }
+
+  AVCaptureConnection *connection = [_captureVideoOutput connectionWithMediaType:AVMediaTypeVideo];
+
+  connection.preferredVideoStabilizationMode = avMode;
+
+  completion(nil);
+}
+
+- (bool)isVideoStabilizationModeSupported:(FCPPlatformVideoStabilizationMode)mode {
+  AVCaptureVideoStabilizationMode avMode = getAvCaptureVideoStabilizationMode(mode);
+
+  return [self isVideoStabilizationAvModeSupported:avMode];
```

Comment:
can these 2 functions be merged

Reply:
This, and other comments you made are a result of me quickly having learned Objective C with the purpose of adding this feature, though without realising the differences in null pointer handling between Objective C and C and handling null pointers like I would with the latter. 

(bool)isVideoStabilizationAvModeSupported:(AVCaptureVideoStabilizationMode)avMode existed only because it had the null safety check and was being reused by (bool)isVideoStabilizationModeSupported:(FCPPlatformVideoStabilizationMode)mode and (void)setVideoStabilizationMode:(FCPPlatformVideoStabilizationMode)mode withCompletion:(void (^)(FlutterError *_Nullable))completion.

I just removed (bool)isVideoStabilizationAvModeSupported:(AVCaptureVideoStabilizationMode)avMode and simplified everything a lot. Thanks!